### DISCUSSION TITLE
Could not find a package "LEMON" in build

### DISCUSSION THREAD

**Discussion by heaton56** [The-OpenROAD-Project/OpenROAD-flow-scripts#632] [General] [answered]
Installing OpenRoad-flow-scripts using the instructions on:
https://openroad.readthedocs.io/en/latest/user/BuildLocally.html

there is a CMake error while running >sudo ./build_openroad.sh --local

The *.cmake files do not seem to be on my system. 

Is there something I am missing here?

Here is the terminal output:

-- SSTA: 0
-- STA executable: /home/robert/OpenROAD-flow-scripts/tools/OpenROAD/src/sta/app/sta
-- GPU is not enabled
CMake Error at src/dpo/CMakeLists.txt:42 (find_package):
  Could not find a package configuration file provided by "LEMON" with any of
  the following names:

    LEMONConfig.cmake
    lemon-config.cmake
    lemonConfig.cmake
    lemon-config.cmake

  Add the installation prefix of "LEMON" to CMAKE_PREFIX_PATH or set
  "LEMON_DIR" to a directory containing one of the above files.  If "LEMON"
  provides a separate development package or SDK, be

 sure it has been
  installed.

-- Configuring incomplete, errors occurred!
See also "/home/robert/OpenROAD-flow-scripts/tools/OpenROAD/build/CMakeFiles/CMakeOutput.log".
See also "/home/robert/OpenROAD-flow-scripts/tools/OpenROAD/build/CMakeFiles/CMakeError.log".

[CMakeError.log](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/9924719/CMakeError.log)
[CMakeOutput.log](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/9924720/CMakeOutput.log)

**Comment by maliberty** [comment]
@vvbandeira please help out

**Reply by rovinski** [reply]
@heaton56 have you installed the OpenROAD dependencies yet? You can use `tools/OpenROAD/etc/DependencyInstaller.sh` to install the required dependencies.

**Comment by heaton56** [comment]
Thanks for your email. The feedback is much appreciated.

Yes I did run DependencyInstaller.sh, and it ran ok.

I am installing from the OpenROAD-flow-scripts project, and using the 
instructions there.

**Comment by heaton56** [comment, marked as answer]
I note from 
https://github.com/The-OpenROAD-Project/OPENROAD_USERS_READ_ME_FIRST/issues/3
that the lemon dependancy is not included in the DependencyInstaller.sh. 

Does this application need to be manually build and configured?

**Reply by vvbandeira** [reply]
Lemon is installed by the script, see [here](https://github.com/The-OpenROAD-Project/OpenROAD/blob/5d135d534695a5bb28321239501abe8adef00dfd/etc/DependencyInstaller.sh#L87-L98).

In the issue you are referring, the user confused lemon (an application that can be installed by `apt`) with lemon (the library we use and install via source code).

Can you check if you have installed lemon via `apt` and try to remove? `apt remove lemon`.
Then re-run the installer script (save the output just in case and upload here if there are still errors) and check if `/usr/local/include/lemon/config.h` exists.

**Comment by heaton56** [comment]
Thanks that worked. 

However, this has effected my system such that most installed apps do not work. 

For example if I type "vim" and return, it hangs. If I then ctrl-c, vim runs. 

I am running ubuntu under wsl, is this is known issue?

**Reply by vvbandeira** [reply]
Good to hear that you were able to compile OpenROAD.

We have not seen such problem. This is a very weird issue, the script only installs packages and does not change any system/user config. I would suggest you check your `.bashrc` / PATH variable and other environment settings. All these are unrelated to OpenROAD and the script.
